neo/VM/Helper.cs

```diff
@@ -228,5 +247,67 @@ private static ContractParameter ToParameter(StackItem item, List<Tuple<StackIte
             return parameter;
         }
+
+        public static StackItem ToStackItem(this ContractParameter parameter)
+        {
+            return ToStackItem(parameter, null);
+        }
+
+        private static StackItem ToStackItem(ContractParameter parameter, List<Tuple<StackItem, ContractParameter>> context)
+        {
+            StackItem stackItem = null;
+            switch (parameter.Type)
+            {
+                case ContractParameterType.Array:
+                    if (context is null)
+                        context = new List<Tuple<StackItem, ContractParameter>>();
+                    else
+                        stackItem = context.FirstOrDefault(p => ReferenceEquals(p.Item2, parameter))?.Item1;
+                    if (stackItem is null)
+                    {
+                        stackItem = ((IList<ContractParameter>)parameter.Value).Select(p => ToStackItem(p, context)).ToList();
+                        context.Add(new Tuple<StackItem, ContractParameter>(stackItem, parameter));
+                    }
+                    break;
+                case ContractParameterType.Map:
+                    if (context is null)
+                        context = new List<Tuple<StackItem, ContractParameter>>();
+                    else
+                        stackItem = context.FirstOrDefault(p => ReferenceEquals(p.Item2, parameter))?.Item1;
+                    if (stackItem is null)
+                    {
+                        stackItem = new Map(((IList<KeyValuePair<ContractParameter, ContractParameter>>)parameter.Value).ToDictionary(p => ToStackItem(p.Key, context), p => ToStackItem(p.Value, context)));
+                        context.Add(new Tuple<StackItem, ContractParameter>(stackItem, parameter));
+                    }
+                    break;
+                case ContractParameterType.Boolean:
+                    stackItem = (bool)parameter.Value;
+                    break;
+                case ContractParameterType.ByteArray:
+                case ContractParameterType.Signature:
+                    stackItem = (byte[])parameter.Value;
+                    break;
+                case ContractParameterType.Integer:
+                    stackItem = (BigInteger)parameter.Value;
+                    break;
+                case ContractParameterType.Hash160:
+                    stackItem = ((UInt160)parameter.Value).ToArray();
+                    break;
+                case ContractParameterType.Hash256:
+                    stackItem = ((UInt256)parameter.Value).ToArray();
+                    break;
+                case ContractParameterType.PublicKey:
+                    stackItem = ((ECPoint)parameter.Value).EncodePoint(true);
+                    break;
+                case ContractParameterType.String:
+                    stackItem = (string)parameter.Value;
+                    break;
+                case ContractParameterType.InteropInterface:
+                    break;
```

Comment:
Should throw like `default`? (I think YES)

Reply:
Shall we also change this?
![image](https://user-images.githubusercontent.com/4393234/64579576-f2f20780-d3b5-11e9-9bee-279695522805.png)

Reply:
I think that we can't create a interop. So my proposal is

```csharp
case InteropInterface _:
case default: throw new ArgumentException();
```

Reply:
If a script returns the interop result, then calling the RPC method ```invokescript```  won't get gas consumed.

Reply:
Ok, use this https://github.com/neo-project/neo/pull/1026#discussion_r322530904 but please, add one unit test for test this behaviour

Reply:
👌